src/Service/ActivityPub/Webfinger/WebFingerParameters.php

```diff
@@ -4,24 +4,25 @@
 
 namespace App\Service\ActivityPub\Webfinger;
 
-use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Request;
 
 class WebFingerParameters
 {
     public const REL_KEY_NAME = 'rel';
     public const HOST_KEY_NAME = 'host';
     public const ACCOUNT_KEY_NAME = 'account';
 
-    public function __construct(private readonly RequestStack $requestStack)
+    public function __construct()
     {
     }
 
-    public function getParams(): array
+    /**
+     * @return array{acccount?: string, host?: string, rel?: array}
+     */
```

Comment:
Maybe use `#[ArrayShape(...)]` instead of this return type? Or is that a PHPStorm only thing?

Reply:
`#[ArrayShape()]` (and a bunch of others) are [phpstorm specific attributes](https://blog.jetbrains.com/phpstorm/2020/10/phpstorm-2020-3-eap-4/), and I didn't use phpstorm so using `ArrayShape` doesn't really do anything[^1], but what I'm using can pick up these doc annotations and use them in offering autocomplete/typing just fine (and imo if phpstorm is actually worth any salt then it should also be able to pick this annotation up too)

[^1]: in fact, it just result in a red diagnostics in my editor for missing attribute class, though it looks like that can be remedied by adding [this package](https://github.com/JetBrains/phpstorm-attributes)